kale/loaddata/dataset_access.py

```diff
@@ -28,21 +28,53 @@ def get_train(self):
         """
         raise NotImplementedError()
 
-    def get_train_val(self, val_ratio):
+    def get_train_class_subset(self, class_ids=None):
+        """
+        Args:
+            class_ids (list, optional): List of chosen subset of class ids.
+        Returns: a torch.utils.data.Dataset
+            Dataset: a torch.utils.data.Dataset with only classes in class_ids
+        """
+        return self._get_subset(self.get_train(), class_ids)
+
+    def get_train_val(self, val_ratio, class_ids=None):
         """
         Randomly split a dataset into non-overlapping training and validation datasets.
 
         Args:
             val_ratio (float): the ratio for validation set
-
+            class_ids (list, optional): List of chosen subset of class ids.
         Returns:
             Dataset: a torch.utils.data.Dataset
         """
-        train_dataset = self.get_train()
+        train_dataset = self.get_train_class_subset(class_ids)
```

Comment:
We may give a flag `class_ids` to `get_train()` directly without changing `self.get_train()` here.